### ISSUE TITLE
Array comparisons always result in `polyEqual` warning

### ISSUE THREAD

**Issue by Skyb0rg007** [smlnj/legacy#331]
### Version

110.99.6.1 (Latest)

### Operating System

- [X] Any
- [ ] Linux
- [ ] macOS
- [ ] Windows
- [ ] Other Unix

### Processor

- [X] Any
- [ ] Arm (using Rosetta)
- [ ] PowerPC
- [ ] Sparc
- [ ] x86 (32-bit)
- [ ] x86-64 (64-bit)
- [ ] Other

### System Component

Core system

### Severity

Cosmetic

### Description

The `polyEqual` warning is meant to warn users that the compiler is falling back to a non-synthesized equality test.
However, this warning occurs on completely legitimate uses on the `array` type.

### Transcript

```sml
- val arr : char array = Array.array (10, #"x");
- arr = arr;
stdIn:2.5 Warning: calling polyEqual
val it = true : bool
```

### Expected Behavior

```sml
- val arr : char array = Array.array (10, #"x");
- arr = arr;
val it = true : bool
```

This should probably not produce a warning even if the type is polymorphic, just like how `ref` works.

```sml
- fun myArrEq (a1 : 'a array, a2 : 'a array) : bool = a1 = a2;
val it = fn : 'a array * 'a array -> bool
```

### Steps to Reproduce

Write `=` on arrays.

### Additional Information

This is probably fairly easy to special-case for now, but the `polyEqual` warning needs an overhaul.
The warning should really be based on the *type* of expressions (ie. the warning should occur before the backend), rather than the contents of the optimized intermediate language.
We should be able to guarantee that all well-understood uses of `op =` are specialized.

### Email address

skyler DOT soss AT gmail DOT com

**Comment by JohnReppy** [comment]
The warning message just means that it is using the generic equality testing function, instead of a type-specialized test.

**Comment by JohnReppy** [comment]
To address this issue, see the `atomeq` function in `base/compiler/FLINT/trans/pequal.sml`; there need to be cases added to handle array types.

**Comment by JohnReppy** [comment]
Fixed for 110.99.7.
